katdal/datasources.py

```diff
@@ -131,18 +135,50 @@ def name(self):
         return name
 
 
+def view_capture_stream(telstate, capture_block_id=None, stream_name=None):
+    """Create telstate view based on detected capture block ID and stream name."""
+    # Detect the capture block
+    if not capture_block_id:
+        capture_blocks = []
+        if 'sdp_capture_block_id' in telstate:
+            for value_time in telstate.get_range('sdp_capture_block_id', st=0):
+                cbid = value_time[0]
+                if 'obs_params' in telstate.view(cbid, exclusive=True):
+                    capture_blocks.append(cbid)
+        if not capture_blocks:
+            raise ValueError('No capture block IDs found in telstate - '
+                             'please specify it manually')
+        # Pick the latest capture block
+        capture_block_id = capture_blocks[-1]
+        if len(capture_blocks) > 1:
+            logger.warn('Telstate has more than one capture block - {} - '
+                        'picking the latest'.format(capture_blocks))
+    # Detect the captured stream
+    if not stream_name:
+        streams = []
+        for stream in telstate['sdp_config']['outputs']:
+            capture_stream = telstate.SEPARATOR.join((capture_block_id, stream))
```

Comment:
I would still check the type of the stream as recorded in sdp_config, to make sure that it's an sdp.l0 stream.

I think we also need to have a face-to-face chat (probably with @sratcliffe) about how this will all work with the flags stream. At the moment it's not explicitly configured as a separate output in sdp_config - it's just on the side of the cal output.

Reply:
I did the check and then decided it's probably superfluous and removed it, as anything with "chunk_info" will be katdal-related. But I could add it back in again.

Reply:
For now. But who knows, maybe someone will be inspired and start writing beamformer data in chunks :-)

Reply:
Done.